pkg/skaffold/sync/sync.go

```diff
@@ -128,21 +129,43 @@ func intersect(context string, syncMap map[string]string, files []string, workin
 			return nil, errors.Wrapf(err, "changed file %s can't be found relative to context %s", f, context)
 		}
 		var matches bool
-		for p, dst := range syncMap {
```

Comment:
i dont like the fact that this for loop is so long and it gets difficult to read.
How about we first find all the TripleStar Patterns, match them and then try all the others?

```

tripleStarSyncMaps, otherSyncMaps = segregateSyncMaps(syncMaps)
for _, f := range files {
 ...
 var match bool
 // first try all tripleStarSyncMaps
 match, dst, err := matchesTripleStarSyncMap(f)
 if err != nil {
    return nil, err
 }
 if match {
   ret[f] = dst
   continue
 }
 // Try matching others
 match, dst, err := matchOthers(otherSyncMaps, f)
 if !match {
  logrus.Infof("Changed file %s does not match any sync pattern. Skipping sync", relPath)
  return nil, nil
 }
 ret[f] = dst
}
...
}


func segregateSyncMaps(syncMap map[string]string) (map[string]string, map[string] string) {
  for p, dst := range syncMap {
     pat := strings.Replace(p, "***", "**", -1)
     if len(pat) != len(p) {
       tripleStarPatter[pat] = dst
     } else {
      doubelStarPattern[p] = dst
    } 
 } 
 return tripleStarPattern, doubleStarPatter
}
```

That ways, we individually test smaller functions.

Reply:
Good suggestion.  Thanks!